client/src/app/pages/environment-deployment-history/environment-deployment-history.component.html

```diff
@@ -17,42 +34,66 @@
         </tr>
       </ng-template>
     </p-table>
-  } @else if (deploymentsQuery.data()?.length === 0) {
+  } @else if (activityHistoryQuery.data()?.length === 0) {
     <!-- Empty State -->
-    <div class="text-gray-500">No deployments found.</div>
+    <div class="text-gray-500">No deployments and locking found.</div>
   } @else {
     <!-- Data Table -->
-    <p-table [value]="deployments() || []" styleClass="p-datatable-sm">
+    <p-table [value]="activityHistory() || []" styleClass="p-datatable-sm">
       <ng-template pTemplate="header">
         <tr>
-          <th>Name</th>
-          <th>Commit SHA</th>
-          <th>State</th>
-          <th>Deployed On</th>
+          <th>Type</th>
+          <th>Description</th>
+          <th>Timestamp</th>
         </tr>
       </ng-template>
-      <ng-template pTemplate="body" let-deployment>
-        <tr>
+      <ng-template pTemplate="body" let-combinedHistory>
+        <tr [class.lock-event]="combinedHistory.type === 'LOCK_EVENT'">
```

Comment:
nitpick/best-practice: You can also use the tailwind syntax here: 
`[class]="combinedHistory.type === 'LOCK_EVENT' ? 'classes-when-condition-is-true' : 'classes-when-conditions-is-false'"`